src/AutoRest.CSharp/Common/Output/Models/ClientMethodBodyLines.cs

```diff
@@ -0,0 +1,216 @@
+                localFunctionName = new CodeWriterDeclaration("FirstPageRequest");
+                return new OneLineLocalFunction(localFunctionName, new[]{KnownParameters.PageSizeHint}, typeof(HttpMessage), requestMethodCall);
+            }
+
+            public static MethodBodyLine NextPageRequest(ValueExpression? restClient, string methodName, IEnumerable<ValueExpression> arguments, out CodeWriterDeclaration localFunctionName)
+            {
+                var requestMethodCall = new InstanceMethodCallExpression(restClient, methodName, arguments.ToList(), false);
+                localFunctionName = new CodeWriterDeclaration("NextPageRequest");
+                return new OneLineLocalFunction(localFunctionName, new[]{KnownParameters.PageSizeHint, KnownParameters.NextLink}, typeof(HttpMessage), requestMethodCall);
+            }
+
+            public static MethodBodyLine Message(ValueExpression value, out CodeWriterDeclaration message)
+            {
+                message = new CodeWriterDeclaration("message");
+                return new UsingDeclareVariableLine(typeof(HttpMessage), message, value);
+            }
+
+            public static MethodBodyLine Response(CSharpType responseType, ValueExpression value, out CodeWriterDeclaration response)
+            {
+                response = new CodeWriterDeclaration("response");
+                return new DeclareVariableLine(responseType, response, value);
+            }
+
+            public static MethodBodyLine RequestContext(ValueExpression value, out CodeWriterDeclaration requestContext)
+            {
+                requestContext = new CodeWriterDeclaration(KnownParameters.RequestContext.Name);
+                return new DeclareVariableLine(KnownParameters.RequestContext.Type, requestContext, value);
+            }
+
+            public static MethodBodyLine NewModelInstance(ModelTypeProvider model, IEnumerable<Parameter> arguments, out CodeWriterDeclaration variable)
+                => NewModelInstance(model, arguments.Select(p => new ParameterReference(p)).ToList(), out variable);
+
+            public static MethodBodyLine NewModelInstance(ModelTypeProvider model, IReadOnlyList<ValueExpression> arguments, out CodeWriterDeclaration variable)
+            {
+                variable = new CodeWriterDeclaration(model.Type.Name.ToVariableName());
+                var newInstance = new NewInstanceExpression(model.Type, arguments);
+                return new DeclareVariableLine(model.Type, variable, newInstance);
+            }
+        }
+    }
+
+    internal record Method(MethodSignature Signature, MethodBody Body);
+    internal record MethodBody(IReadOnlyList<MethodBodyBlock> Blocks);
+
+    internal record MethodBodyBlock
+    {
+        public static MethodBodyBlock Create(params MethodBodyLine[] lines) => new MethodBodyLines(lines);
+    }
+
+    internal record ParameterValidationBlock(IReadOnlyList<Parameter> Parameters) : MethodBodyBlock;
```

Comment:
Can we please have each class represented by a file, its very difficult especially in github pr review to figure out where these things are defined.

Reply:
Sure.